### PR TITLE
Fixes issue #90: added support for CPPcheck result containing file "n…

### PR DESCRIPTION
**Pull request by Hansvdsteen**
…ofile" and line-number "0"

### CODE REVIEW

**Comment by Hansvdsteen** [comment]
A have opened a ticket at cppcheck itself to ask if this behaviour is kept or not, but for now this seems te best solution.

**Comment by kwin** [comment]
@Hansvdsteen Thanks a lot for the PR. Can you reference the ticket over at cppcheck as well?

File: com.googlecode.cppcheclipse.core/src/com/googlecode/cppcheclipse/core/command/CppcheckCommand.java [hunk 1]
```diff
@@ -297,14 +297,14 @@ public static Problem parseResult(String line, IProject project) {
 		try {
 
 			File filename;
-			if (Strings.isNullOrEmpty(lineParts[0])) {
+			if (Strings.isNullOrEmpty(lineParts[0]) || lineParts[0].equals(new String("nofile"))) {
```
Comment (kwin) [inline com.googlecode.cppcheclipse.core/src/com/googlecode/cppcheclipse/core/command/CppcheckCommand.java]: any reason why "nofile".equals(lineParts[0]) is not good enough (that will never throw an NPE and does not require calling the string constructor)?

**Comment by Hansvdsteen** [comment]
#8866
“Cppcheck 1.84 and later, result containing file "nofile" and "line-number" 0 breaks external eclipse plugin”

^Hans van der Steen

File: com.googlecode.cppcheclipse.core/src/com/googlecode/cppcheclipse/core/command/CppcheckCommand.java (hunk 1 above)
Comment (Hansvdsteen) [reply com.googlecode.cppcheclipse.core/src/com/googlecode/cppcheclipse/core/command/CppcheckCommand.java]: Not really.. but if it works its also fine by me... i change/test this.

Comment (Hansvdsteen) [reply com.googlecode.cppcheclipse.core/src/com/googlecode/cppcheclipse/core/command/CppcheckCommand.java]: Please take another look at it..
Other question.. when is a new version released (including this fix if accepted)

**Comment by kwin** [comment]
Merged now. Thanks a lot for the contribution @Hansvdsteen. Don't know when I will spin the next release though.

**Comment by Hansvdsteen** [comment]
@kwin, would be nice to make a release in short time. Its already some time ago (2016) and this issue is really a blocking one for me..

**Pull request merged**